src/coreclr/pal/src/exception/machexception.cpp

```diff
@@ -1148,8 +1156,25 @@ SEHExceptionThread(void *args)
 
             if (!feFound)
             {
-                NONPAL_TRACE("HijackFaultingThread thread %08x\n", thread);
-                HijackFaultingThread(thread, mach_task_self(), sMessage);
+#ifdef HOST_ARM64
+                // Check for a special case of invalid instruction that is used to implement RestoreCompleteContext
```

Comment:
GCStress=C is under `HAVE_GCCOVER` that is only ever enabled in debug builds. This special treatment can be under debug ifdef if we wanted to reduce amount of work done in release builds.

Reply:
@jkotas  I've decided to redo it a bit differently. I have moved creation of the `MachExceptionInfo` out of the `HijackFaultingThread` to the caller. That way we don't need additional call to get the thread state.